quolance-api/src/main/java/com/quolance/quolance_api/services/business_workflow/impl/FreelancerWorkflowServiceImpl.java

```diff
@@ -123,6 +123,12 @@ public List<ProjectPublicDto> getAllAvailableProjects() {
 
     @Override
     public ProjectPublicDto getProject(Long projectId) {
+        if(ProjectPublicDto.fromEntity(projectService.getProjectById(projectId)).getProjectStatus().equals(ProjectStatus.PENDING)){
```

Comment:
get project once to use in your check and the return statement.

make check also about the visibility date. something like:
`project.getVisibilityExpirationDate().isBefore(currentDate))`

should not see a project after the visibility date